contract_payment_mode/models/contract.py

```diff
@@ -13,7 +13,10 @@ class ContractContract(models.Model):
 
     @api.onchange("partner_id")
     def on_change_partner_id(self):
-        self.payment_mode_id = self.partner_id.customer_payment_mode_id.id
+        if self.contract_type == "purchase":
+            self.payment_mode_id = self.partner_id.supplier_payment_mode_id.id
```

Comment:
Now that you are here, let's improve this for being aware of the company per document feature:

```suggestion
            self.payment_mode_id = self.with_company(self.company_id).partner_id.supplier_payment_mode_id.id
```

Reply:
@pedrobaeza Done